src/main/java/org/broadinstitute/consent/http/service/DatabaseMatchProcessAPI.java

```diff
@@ -41,10 +40,10 @@ public void processMatchesForConsent(String consentId) {
     @Override
     public void processMatchesForPurpose(String purposeId) {
         removeMatchesForPurpose(purposeId);
-        Document rp = dataAccessRequestService.getDataAccessRequestByReferenceIdAsDocument(purposeId);
-        if (rp != null && rp.get(DarConstants.RESTRICTION) != null) {
-            Match match = matchingServiceAPI.findMatchForPurpose(purposeId);
-            saveMatch(Arrays.asList(match));
+        DataAccessRequest dar = dataAccessRequestService.findByReferenceId(purposeId);
+        if (Objects.nonNull(dar)) {
+            Match match = matchingServiceAPI.findMatchForPurpose(dar.getReferenceId());
+            saveMatch(Collections.singletonList(match));
         }
```

Comment:
Simplified to remove `DarConstants.RESTRICTION`
Also removes a dependency on `Document` which we need to do as general cleanup.